lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -743,6 +743,61 @@ func toClient(handle C.uintptr_t) (value *Client, err error) {
 	return cgo.Handle(handle).Value().(*Client), nil
 }
 
+//export cgo_read_rdp_license
+func cgo_read_rdp_license(handle C.uintptr_t, req *C.CGOLicenseRequest, data_out **C.uint8_t, len_out *C.uint32_t) C.CGOErrCode {
+	client, err := toClient(handle)
+	if err != nil {
+		return C.ErrCodeFailure
+	}
+
+	issuer := C.GoString(req.issuer)
+	company := C.GoString(req.company)
+	productID := C.GoString(req.product_id)
+
+	license, err := client.readRDPLicense(
+		uint16(req.major_version), uint16(req.minor_version),
+		issuer, company, productID)
+	if err != nil {
+		*data_out = nil
```

Comment:
```suggestion
		*data_out = nil
		*len_out = 0
```

Reply:
I'd move this early in the function so unless we are returning values we are at least clearing out our out params.